operate/common/src/main/java/io/camunda/operate/connect/ElasticsearchConnector.java

```diff
@@ -186,27 +194,76 @@ public RestHighLevelClient createEsClient(ElasticsearchProperties elsConfig) {
   }
 
   protected HttpAsyncClientBuilder configureHttpClient(
-      HttpAsyncClientBuilder httpAsyncClientBuilder, ElasticsearchProperties elsConfig) {
+      final HttpAsyncClientBuilder httpAsyncClientBuilder,
+      final ElasticsearchProperties elsConfig) {
     setupAuthentication(httpAsyncClientBuilder, elsConfig);
+
+    LOGGER.trace("Attempt to load interceptor plugins");
+    if (elsConfig.getInterceptorPlugins() != null) {
+      loadInterceptorPlugins(httpAsyncClientBuilder, elsConfig);
+    }
+
     if (elsConfig.getSsl() != null) {
       setupSSLContext(httpAsyncClientBuilder, elsConfig.getSsl());
     }
     return httpAsyncClientBuilder;
   }
 
+  private void loadInterceptorPlugins(
+      final HttpAsyncClientBuilder httpAsyncClientBuilder,
+      final ElasticsearchProperties elsConfig) {
+    LOGGER.trace("Plugins detected to be not empty {}", elsConfig.getInterceptorPlugins());
+
+    final List<InterceptorPluginProperty> interceptors = elsConfig.getInterceptorPlugins();
+    for (final InterceptorPluginProperty interceptor : interceptors) {
+      LOGGER.trace("Attempting to register {}", interceptor.getId());
+      try {
+        // WARNING! Due to the nature of interceptors, by the moment they (interceptors)
+        // are executed, the below class loader will close the JAR file hence
+        // to avoid NoClassDefFoundError we must not close this class loader.
```

Comment:
:+1: Nice catch. I would add this information on the `ExternalJarClassLoader` java doc - we could even overload the `close` implementation of that class and log a warning there, e.g. `Closing the external JAR class loader may lead to ClassNotFoundException; only close if it's truly necessary`.

Reply:
I thought to add it in the follow up PR but added in this one